### PR TITLE
Internal Solution to DMOJ-Phidias in Intro to DP, Gold

### PR DESCRIPTION
**Pull request by Qw3rTy827**

### CODE REVIEW

File: solutions/dmoj-phidias.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+
+#include <bits/stdc++.h>
+#define pi pair<int,int>
+#define fi first
+#define se second
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:69]: the convention is to use just `f` and `s`

File: solutions/dmoj-phidias.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+using namespace std;
+
+const int maxW = 605;
+const int maxN = 205;
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:73]: `MAX_W` and `MAX_N` instead
also 600 and 200 should work, no? can you change it to that?

File: solutions/dmoj-phidias.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
+---
+id: dmoj-phidias
+source: DMOJ
+title: phidias
+author: Andrew Cheng
+---
+
+## Solution
+
+Let $f[i][j]$ be the amount of waste material for a rectangle with width $i$ and height $j$. This rectangle of width $i$ and height $j$ can be split into two smaller rectangles (marked in red and green) after removing a plate that Phidias wants (marked in blue). There are two different possible ways to split the remaining shape into two rectangles.
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:10]: maybe just use `dp` (and change the code along with it) to be consistent?

File: solutions/dmoj-phidias.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+pi a[maxN];
+int f[maxW][maxW];
+int N,W,H;
+
+signed main(){
+	ios_base::sync_with_stdio(false);
+	cin.tie(0);
+	cin >> W >> H >> N;
+	for(int i = 1; i <= N; ++i)cin >> a[i].fi >> a[i].se;
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:83]: spaces for the control statements and stuff please

File: solutions/dmoj-phidias.mdx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+#define fi first
+#define se second
+using namespace std;
+
+const int maxW = 605;
+const int maxN = 205;
+
+pi a[maxN];
+int f[maxW][maxW];
+int N,W,H;
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:77]: 1. move this into `main`
2. put spaces between the commas

File: solutions/dmoj-phidias.mdx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+signed main(){
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:79]:
```suggestion
int main() {
```

File: solutions/dmoj-phidias.mdx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+		for(int j = 1; j <= H; ++j){
+			f[i][j] = i * j;
+		}
+	}
+	for(int i = 1; i <= N; ++i)f[a[i].fi][a[i].se] = 0;
+	for(int i = 1; i <= W; ++i){
+		for(int j = 1; j <= H; ++j){
+			for(int k = 1; k <= N; ++k){
+				if(i < a[k].fi || j < a[k].se)continue;
+				f[i][j] = min(f[i][j],f[i][j-a[k].se] + f[i-a[k].fi][a[k].se]);
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:94]: spaces between ops

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/dmoj-phidias.mdx (hunk 2 above)
Comment (Qw3rTy827) [reply solutions/dmoj-phidias.mdx:73]: umm W and H can reach 600 and the same applies for N and 200. So I don't think an array size of exactly 600 / 200 would work.

File: solutions/dmoj-phidias.mdx (hunk 5 above)
Comment (Qw3rTy827) [reply solutions/dmoj-phidias.mdx:77]: Why would I move my arrays and variables into ```main``` though ?

Comment (SansPapyrus683) [reply solutions/dmoj-phidias.mdx:77]: just the variables
not the arrays

File: solutions/dmoj-phidias.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/dmoj-phidias.mdx:73]: why not?

Comment (Qw3rTy827) [reply solutions/dmoj-phidias.mdx:73]: I start my arrays from 1 not 0. I mean I don't think using a couple more bytes of memory would be a big problem.

Comment (SansPapyrus683) [reply solutions/dmoj-phidias.mdx:73]: then just still set them to 600 and 200, and put `MAX_W + 1` in your array declarations

File: solutions/dmoj-phidias.mdx [hunk 8]
```diff
@@ -0,0 +1,108 @@
+---
+id: dmoj-phidias
```
Comment (dolphingarlic) [inline solutions/dmoj-phidias.mdx:2]: Source should be changed to IOI 2004

Comment (dolphingarlic) [reply solutions/dmoj-phidias.mdx:2]: Also why is this problem even in the guide. It's not that nice

File: solutions/dmoj-phidias.mdx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+using namespace std;
+
+const int maxW = 605;
+const int maxN = 205;
+
+pi a[maxN];
+int dp[maxW][maxW];
+int N,W,H;
+
+int main(){
```
Comment (jessechoe10) [inline solutions/dmoj-phidias.mdx:79]: Comment your code bruh

File: solutions/dmoj-phidias.mdx (hunk 8 above)
Comment (jessechoe10) [reply solutions/dmoj-phidias.mdx:2]: Yeah... Why is it still here?

Comment (Qw3rTy827) [reply solutions/dmoj-phidias.mdx:2]: The problem ID and source in the contents folder is still "dmoj-phidias" and "dmoj" though. Do you want me to change those as well ?

File: solutions/dmoj-phidias.mdx (hunk 9 above)
Comment (Qw3rTy827) [reply solutions/dmoj-phidias.mdx:79]: can you be a bit more specific about the parts that is hard to understand / needs extra commenting ?

File: solutions/dmoj-phidias.mdx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+```cpp
+
+#include <bits/stdc++.h>
+#define pi pair<int,int>
+#define f first
+#define s second
+using namespace std;
+
+const int maxW = 600;
+const int maxN = 200;
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:73]: rename to MAX_W and MAX_N respectively

File: solutions/dmoj-phidias.mdx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+int dp[maxW+1][maxW+1];
+int N,W,H;
+
+int main(){
+	ios_base::sync_with_stdio(false);
+	cin.tie(0);
+	cin >> W >> H >> N;
+	for(int i = 1; i <= N; ++i){
+        cin >> a[i].f >> a[i].s;
+    }
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:85]: indentation seems hecked up

File: solutions/dmoj-phidias.mdx (hunk 9 above)
Comment (SansPapyrus683) [reply solutions/dmoj-phidias.mdx:79]: the dp part probably

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/dmoj-phidias.mdx (hunk 8 above)
Comment (dolphingarlic) [reply solutions/dmoj-phidias.mdx:2]: Yes please

Comment (dolphingarlic) [reply solutions/dmoj-phidias.mdx:2]: Or rather just remove the problem (I don't like it)

File: solutions/dmoj-phidias.mdx [hunk 12]
```diff
@@ -91,10 +93,13 @@ int main(){
 	for(int i = 1; i <= N; ++i){
         dp[a[i].f][a[i].s] = 0;
     }
+    
 	for(int i = 1; i <= W; ++i){
 		for(int j = 1; j <= H; ++j){
 			for(int k = 1; k <= N; ++k){
+                //If the rectangle is not big enough
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:100]: indentation seems hecked up

File: solutions/dmoj-phidias.mdx [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+```cpp
+
+#include <bits/stdc++.h>
+#define pi pair<int,int>
+#define f first
+#define s second
+using namespace std;
+
+const int max_W = 600;
+const int max_N = 200;
```
Comment (SansPapyrus683) [inline solutions/dmoj-phidias.mdx:73]: rename to MAX_W and MAX_N

**Pull request closed without merging**